src/map/utils/charutils.cpp

```diff
@@ -5331,6 +5333,37 @@ namespace charutils
         return 0;
     }
 
+    void AddUnityPoints(CCharEntity* PChar, int32 amount, int32 max)
+    {
+        float       evalPoints    = static_cast<float>(amount) / 1000;
+        const char* rankingQuery  = "UPDATE unity_system SET points_current = points_current+%f WHERE leader=%d;";
+
+        if (amount < 0)
+        {
+            // Do not allow subracting points with this function
+            return;
+        }
+
+        AddPoints(PChar, "unity_accolades", amount, max);
+        AddPoints(PChar, "current_accolades", amount, std::numeric_limits<int32>::max()); // Do not cap current_accolades
+        Sql_Query(SqlHandle, rankingQuery, evalPoints, PChar->profile.unity_leader);
+
+        PChar->pushPacket(new CCharStatsPacket(PChar));
+    }
```

Comment:
How do we feel about rolling this into the existing AddPoints with a condition similar to how we handle `spark_of_eminence`?

Reply:
So long as we don't have a case where we'd want to add accolades to a user without affecting ranking, I'm on board with that.  Any thoughts?

Reply:
You could bundle that in a check couldn't you, like:
```cpp
    void AddPoints(CCharEntity* PChar, const char* type, int32 amount, int32 max)
    {
        const char* Query = "UPDATE char_points SET %s = GREATEST(LEAST(%s+%d, %d), 0) WHERE charid = %u;";

        Sql_Query(SqlHandle, Query, type, type, amount, max, PChar->id);

        if (strcmp(type, "unity_accolades") == 0)
        {
            float       evalPoints    = static_cast<float>(amount) / 1000;
            const char* rankingQuery  = "UPDATE unity_system SET points_current = points_current+%f WHERE leader=%d;";
            AddPoints(PChar, "current_accolades", amount, std::numeric_limits<int32>::max()); // Do not cap current_accolades
            Sql_Query(SqlHandle, rankingQuery, evalPoints, PChar->profile.unity_leader);

            PChar->pushPacket(new CCharStatsPacket(PChar));
        }
        else if (strcmp(type, "spark_of_eminence") == 0)
        {
            PChar->pushPacket(new CRoeSparkUpdatePacket(PChar));
        }
    }
```

Reply:
I think I was thinking more of a case revolving around a gm command; outside of that, there shouldn't be any situation where adding accolades doesn't impact ranking, so I'll make this change.

Reply:
Back to this real quick, do we still need the `addAccolades` binding? I'm just seeing it in the GM command and 1 other spot. Is the difference just the cap? Could use `player:addCurrency("unity_accolades", rewards["accolades"], CAP_CURRENCY_ACCOLADES)`?

Reply:
Agreed.  Since accolades are only granted as a result of RoE objective especially, there's no need for this binding or gm command.  Removed.